superset/security/manager.py

```diff
@@ -170,32 +170,37 @@ def query_context_modified(query_context: "QueryContext") -> bool:
     if form_data.get("slice_id") != stored_chart.id:
         return True
 
-    # compare form_data
-    requested_metrics = {
-        freeze_metric(metric) for metric in form_data.get("metrics") or []
-    }
-    stored_metrics = {
-        freeze_metric(metric)
-        for metric in stored_chart.params_dict.get("metrics") or []
-    }
-    if not requested_metrics.issubset(stored_metrics):
-        return True
+    stored_query_context = (
+        json.loads(cast(str, stored_chart.query_context))
+        if stored_chart.query_context
+        else None
+    )
 
-    # compare queries in query_context
-    queries_metrics = {
-        freeze_metric(metric)
-        for query in query_context.queries
-        for metric in query.metrics or []
-    }
+    # compare columns and metrics in form_data with stored values
+    for key in ["metrics", "columns"]:
```

Comment:
I think some viz types might have the columns listed as `groupby` in the `form_data`:

![image](https://github.com/apache/superset/assets/96086495/3d74e982-69e4-4efc-b679-6b0b04b2b677)

Reply:
I tested via the browser (just inspecting the request the frontend sends to the `/api/v1/chart/data` endpoint, but it's possible this structure changes until it gets to this point, so I could be wrong.

Reply:
I do see it listed as `columns` in the `queries`, tho:

![image](https://github.com/apache/superset/assets/96086495/d4c617e3-30f2-4b54-b5b0-1eeba0033953)